hw01/hw01.py

```diff
@@ -66,6 +65,11 @@ def largest_factor(n):
     1
     """
     "*** YOUR CODE HERE ***"
+    i=n-1
+    while n%i!=0:
```

Comment:
What if n = 0 or negative value? This function will enter into infinity loop....

Reply:
add if n=0

Reply:
What about negative value?  Please, think comprehensively, what is the main error, what potential issue the code may have, and how to solve it properly.

Reply:
if n < 0,it will return -1